src/ducks/account/helpers.js

```diff
@@ -163,43 +163,64 @@ const buildReimbursementsVirtualAccount = (filter, options) => transactions => {
   })
 
   const account = {
-    _id: options.id,
+    _id: specs.id,
     _type: ACCOUNT_DOCTYPE,
-    label: `Data.virtualAccounts.${options.translationKey}`,
+    label: `Data.virtualAccounts.${specs.translationKey}`,
     balance,
     type: 'Reimbursements',
-    categoryId: options.categoryId,
+    categoryId: specs.categoryId,
     currency: '€',
     virtual: true
   }
 
   return account
 }
 
-export const buildHealthReimbursementsVirtualAccount = buildReimbursementsVirtualAccount(
-  isHealthExpense,
-  {
+const reimbursementsVirtualAccountsSpecs = {
+  health: {
     id: 'health_reimbursements',
     translationKey: 'healthReimbursements',
-    categoryId: getCategoryIdFromName('healthExpenses')
+    categoryId: getCategoryIdFromName('healthExpenses'),
+    filter: isHealthExpense
+  },
+  professional: {
+    id: 'professional_reimbursements',
+    translationKey: 'professionalReimbursements',
+    categoryId: getCategoryIdFromName('professionalExpenses'),
+    filter: isProfessionalExpense
   }
+}
+
+export const buildHealthReimbursementsVirtualAccount = buildReimbursementsVirtualAccount(
+  reimbursementsVirtualAccountsSpecs.health
 )
 
 export const buildProfessionalReimbursementsVirtualAccount = buildReimbursementsVirtualAccount(
-  isProfessionalExpense,
-  {
-    id: 'professional_reimbursements',
-    translationKey: 'professionalReimbursements',
-    categoryId: getCategoryIdFromName('professionalExpenses')
-  }
+  reimbursementsVirtualAccountsSpecs.professional
 )
 
+const any = (...fns) => (...args) => {
+  for (const fn of fns) {
+    if (fn(...args)) {
+      return true
+    }
+  }
+}
+
+const othersFilter = transaction => {
+  const filters = Object.values(reimbursementsVirtualAccountsSpecs).map(
+    spec => spec.filter
+  )
+
+  return !any(...filters)(transaction) && isExpense(transaction)
+}
```

Comment:
```
const isSpecificReimbursementFilter = overSome(reimbursementsVirtualAccountsSpecs.map(x => x.filter))
const otherFilter = transaction => !isSpecificReimbursementFilter(transaction) && isExpense(transction)
```

Reply:
Otherwise you're building the filter function for each function that is filtered ;)

Reply:
Thanks https://github.com/cozy/cozy-banks/pull/1773/commits/b04cc102c3debaabc7c2c2e8246a543df67bd358